hana/lib/HANAService.js

```diff
@@ -720,27 +720,61 @@ class HANAService extends SQLService {
       }
 
       // HANA does not allow WHERE TRUE so when the expression is only a single entry "= TRUE" is appended
-      if (caseSuffix && xpr.length === 1) {
+      if (caseSuffix && (
+        xpr.length === 1 || xpr.at(-1) === '')) {
         sql.push(caseSuffix)
       }
 
       return `${sql.join(' ')}`
     }
 
     operator(x, i, xpr) {
+      const up = x.toUpperCase()
       // Add "= TRUE" before THEN in case statements
-      // As all valid comparators are converted to booleans as SQL specifies
-      if (x in { THEN: 1, then: 1 }) return ` = TRUE ${x}`
-      if ((x in { LIKE: 1, like: 1 } && is_regexp(xpr[i + 1]?.val)) || x === 'regexp') return 'LIKE_REGEXPR'
+      if (
+        up in logicOperators &&
+        !this.comparerator({ xpr }, i - 1)
+      ) {
+        return ` = TRUE ${x}`
+      }
+      if (
+        (up === 'LIKE' && is_regexp(xpr[i + 1]?.val)) ||
+        up === 'REGEXP'
+      ) return 'LIKE_REGEXPR'
       else return x
     }
 
+    get is_distinct_from_() { return '!=' }
+    get is_not_distinct_from_() { return '==' }
+
+    /**
+     * Checks if the xpr is a comparison or a value
+     * @param {} xpr 
+     * @returns 
+     */
+    comparerator({ xpr }, start) {
```

Comment:
```suggestion
    comparator({ xpr }, start) {
```
or maybe `is_comparator` based on the docs comment?